lib/tandem/tandemSimulator.js

```diff
@@ -79,6 +79,22 @@ exports.make = function(config){
 
   return {
     alarm: function(event) {
+      if (currStatus !== null && currStatus.status === 'suspended') {
+        var type = event.alarmType;
+        if(type === 'occlusion' || type === 'auto_off' || type === 'no_insulin' || type === 'no_power' ||
+            (type === 'other' && event.payload.alarm_name === 'Cartridge Removed') ) {
+
+          var status = currStatus;
+          status.reason = {suspended: 'automatic'};
+          if(type === 'other') {
+            status.payload = {cause: type, description: event.payload.alarm_name};
```

Comment:
It looks like Jellyfish doesn't store any extra information with the cause in 'suspended' and 'resumed' in the payload, so I'm only going to fill in the suspend reasons for the other events (occlusion, no_power, no_insulin, auto_off). 

It also looks like 'cartridge removed' alarm events arrive before the suspended device event, so would then need a `setCurrAlarm` and all that goes with it (processing alarm clear events etc.). If we're then only storing {cause:'other'}, I feel like it's not worth the effort?